code/Extension/UserFormFileExtension.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace SilverStripe\UserForms\Extension;
+
+use SilverStripe\Assets\File;
+use SilverStripe\Assets\Folder;
+use SilverStripe\ORM\DataExtension;
+use SilverStripe\ORM\DataObject;
+use SilverStripe\ORM\Queries\SQLUpdate;
+use SilverStripe\UserForms\Control\UserDefinedFormController;
+use SilverStripe\UserForms\Model\Submission\SubmittedFileField;
+use SilverStripe\Versioned\Versioned;
+
+class UserFormFileExtension extends DataExtension
+{
+
+    public const USER_FORM_UPLOAD_UNKNOWN = null;
+
+    public const USER_FORM_UPLOAD_FALSE = 'f';
+
+    public const USER_FORM_UPLOAD_TRUE = 't';
+
+    private static $db = [
+        'UserFormUpload' => "Enum('f, t', null)",
```

Comment:
It feels rather strange to me to use an `Enum` data type when the allowed values are restricted by the class constants. I don't think it should be restricted in both the database layer and the application layer in this way.

I'd recommend using a single char varchar or even an bit / tiny int value and use constants that match the type. The alternative is just to use Enums and the built-in functionality around them. However, I would discourage the use of Emuns if possi ble.

Reply:
I see most of the "contraint" happening via the Enum, the class constants are really just to make the code read nicer since we're using 'f'/'t'/null instead of good old boolean's

The values 'f'/'t'/null will never change, so not sure how loosing the Enum to Varchar(1) would be of benefit?

Enum's are used a fair bit elsewhere in the codebase, though there something inherently bad with Enum?